ext/flash/01_http.js

```diff
@@ -188,6 +188,44 @@
     return hostname === "0.0.0.0" ? "localhost" : hostname;
   }
 
+  function writeFixedResponse(
+    server,
+    requestId,
+    response,
+    end,
+    respondFast,
+  ) {
+    let nwritten = 0;
+    // TypedArray
+    if (typeof response !== "string") {
+      nwritten = respondFast(requestId, response, end);
+    } else {
+      // string
+      const maybeResponse = stringResources[response];
+      if (maybeResponse === undefined) {
+        stringResources[response] = core.encode(response);
+        nwritten = core.ops.op_flash_respond(
+          server,
+          requestId,
+          stringResources[response],
+          end,
+        );
+      } else {
+        nwritten = respondFast(requestId, maybeResponse, end);
+      }
+    }
+
+    if (nwritten < response.length) {
```

Comment:
if i understand correctly, when we fail to write full response this will just queue up an async request to write the remaining bytes out later? but, we don't seem to await this async operation or catch any async error afaict.

when this happens, it means we have to wait for socket to become writable before trying to write again - but am not sure where that happens, if at all? this could be just down to me not understand the code all the way down.

Reply:
It is a one-off async op - it will resolve when all the response has been written.